utils/src/main/java/org/apache/cloudstack/utils/redfish/RedfishClient.java

```diff
@@ -0,0 +1,423 @@
+        BASIC_AUTH, SESSION
+    }
+
+    /**
+     * Redfish System Power State: </br>
+     * <b>Off:</b> The state is powered Off. </br>
+     * <b>On:</b> The state is powered On. </br>
+     * <b>PoweringOff:</b> A temporary state between On and Off. </br>
+     * <b>PoweringOn:</b> A temporary state between Off and On.
+     */
+    public enum RedfishPowerState {
+        On, Off, PoweringOn, PoweringOff
+    }
+
+    /**
+     * <ul>
+     * <li><b>ForceOff:</b> Turn the unit off immediately (non-graceful shutdown).
+     * <li><b>ForceOn:</b> Turn the unit on immediately.
+     * <li><b>ForceRestart:</b> Perform an immediate (non-graceful) shutdown,
+     * followed by a restart.
+     * <li><b>GracefulRestart:</b> Perform a graceful shutdown followed by a restart
+     * of the system.
+     * <li><b>GracefulShutdown:</b> Perform a graceful shutdown and power off.
+     * <li><b>Nmi:</b> Generate a Diagnostic Interrupt (usually an NMI on x86
+     * systems) to cease normal operations, perform diagnostic actions and typically
+     * halt the system.
+     * <li><b>On:</b> Turn the unit on.
+     * <li><b>PowerCycle:</b> Perform a power cycle of the unit.
+     * <li><b>PushPowerButton:</b> Simulate the pressing of the physical power
+     * button on this unit.
+     * </ul>
+     */
+    public enum RedfishResetCmd {
+        ForceOff, ForceOn, ForceRestart, GracefulRestart, GracefulShutdown, Nmi, On, PowerCycle, PushPowerButton
+    }
+
+    /**
+     *
+     */
+    public RedfishClient(String username, String password, RedfishAuthType authenticationType, boolean useHttps, boolean ignoreSSL) {
+        this.username = username;
+        this.password = password;
+        this.authType = authenticationType;
+        this.useHttps = useHttps;
+        this.ignoreSSL = ignoreSSL;
+    }
+
+    /**
+     *
+     */
+    private String buildRequestUrl(String hostAddress, RedfishCmdType cmd, String resourceId) {
+        String urlHostAddress = validateAddressAndPrepareForUrl(hostAddress);
+        String requestPath = getRequestPathForCommand(cmd, resourceId);
+
+        if (useHttps) {
+            return String.format("https://%s/%s", urlHostAddress, requestPath);
+        } else {
+            return String.format("http://%s/%s", urlHostAddress, requestPath);
+        }
+    }
+
+    /**
+     *
+     */
+    private HttpResponse executeGetRequest(String url) {
+        try {
+            URIBuilder builder = new URIBuilder(url);
+            builder.setUserInfo(username, password);
+
+            HttpGet httpReq = new HttpGet(builder.build());
+            httpReq.addHeader("accept", "application/json");
+
+            HttpClient client;
+            if (ignoreSSL) {
+                client = ignoreSSLCertValidator();
+            } else {// TODO spawn a regular http client
+                throw new RedfishException("Redfish Client does not support SSL certificates. Please set the client to ignore SSL certificates.");
```

Comment:
Why is this needed? Because you can just spawn a regular HTTP client without anything else. It will then complain if it can't validate the SSL certificate.